### DISCUSSION TITLE
Helm chart improvements

### DISCUSSION THREAD

**Discussion by reubenhEROAD** [featurehub-io/featurehub#995] [Ideas]
I am self hosting featurehub, using terraform to install the helm chart on EKS, and Datadog for observability.

I have had success getting everything running but I did have to make a few modifications to the helm chart. I am wondering if anyone else has encountered these items, if there is any way around them, or if they might be addressed in the future so I can use the provided helm chart instead of self hosting:

- To expose the prometheus metrics to the datadog agent, an annotation with a JSON value needs to be added to the pods, shown below. Helm doesn’t like this and the only way to make it work has been to add `toJson` function to the deployment templates like below. Perhaps a solution could be a seperate `podAnnotationsJson` value which converts annotation values into JSON using the helm function.

```
podAnnotations:
    ad.datadoghq.com/featurehub-dacha.checks: |
      {
        "openmetrics": {
          "init_config": {},
          "instances": [
            {
              "openmetrics_endpoint": "http://featurehub-dacha.featurehub:8701/metrics",
              "namespace": "eks.featurehub.dacha",
              "metrics": [".*"]
            }
          ]
        }
      }
```

```
{{- range $key, $value := .Values.edge.podAnnotations }}
        {{ $key }}: {{ $value | toJson }}
{{- end }}
```

- Enabling `.Values.*.prometheus.enabled` enables the ServiceMonitor resources - this requires installation of the prometheus operator into the cluster and is not needed for my case, so ideally there would be a seperate value to toggle to expose prometheus port on the services and enable the ServiceMonitor
- To correctly tag all of the data collected by Datadog I need to add labels to the deployments and pods. The nats helm chart has values for this `statefulSetPodLabels` & `commonLabels`  - it would be great if the featurehub helm chart could support this
- The ingress load balancers used by EKS only work with ClusterIP services and not NodePort, so ideally would be able to toggle/choose between the two service types
- And finally I think this is a mistake that no one has picked up yet: the edge & MR deployment templates refer to a value `.Values.podAnnotations` when it should be `.Values.managementRepository.podAnnotations` & `.Values.edge.podAnnotations`

Maybe I could be the first person to encounter these issues. I am happy to try contribute to improve the helm chart (please let me know if there is anything I need to consider before making a PR?)
Thanks :slightly_smiling_face:

**Comment by reubenhEROAD** [comment]
The method used by the NATS helm chart appears to handle JSON annotation values fine, so using this way could be an option:
https://github.com/nats-io/k8s/blob/main/helm/charts/nats/templates/statefulset.yaml#LL42C7-L44C17

**Comment by IrinaSouth** [comment]
hi @reubenhEROAD ! Thanks for the detailed suggestion for the helm chart improvement, please go ahead and raise a PR in the Helm repo. Cheers!
